core/shared/src/main/scala/scalaz/zio/internal/Executor.scala

```diff
@@ -143,13 +143,17 @@ object Executor extends Serializable {
       def shutdown(): Unit = ()
     }
 
+  private[zio] trait MeteredExecutor extends Executor {
```

Comment:
Rather than using a subtype, how about we make this `Option[ExecutionMetrics]` on `Executor`? The test can require it be `Some`.

Reply:
Done. I thought metrics were not exposed intentionally. They will be publicly accessible now.